core/src/components/cat-radio/cat-radio.tsx

```diff
@@ -0,0 +1,89 @@
+import { Component, Event, EventEmitter, h, Prop } from '@stencil/core';
+
+let nextUniqueId = 0;
+/**
+ *
+ * @part radio - The input type radio element.
+ * @part label - The label of the input.
+ */
+@Component({
+  tag: 'cat-radio',
+  styleUrl: 'cat-radio.scss',
+  shadow: true
+})
+export class CatRadio {
+  private readonly id = `cat-radio-${++nextUniqueId}`;
+
+  /**
+   * Whether this radio is checked.
+   * */
+  @Prop() checked?: boolean;
+
+  /**
+   * Whether the radio is required.
+   * */
+  @Prop() required?: boolean;
+
+  /**
+   *  Whether this radio is disabled.
+   *  */
+  @Prop() disabled = false;
+
+  /**
+   * The name of the radio component.
+   */
+  @Prop() name?: string;
+
+  /**
+   * The value of the radio component.
+   */
+  @Prop() value?: string;
+
+  /**
+   * The label of the radio that is visible.
+   */
+  @Prop() label?: string;
```

Comment:
We want to put accessibility first, so one of the approaches I took was that components actively complain if they are not accessible. So what do you think of logging an error if no label is set (because it always needs to be set, even if it is hidden). Take a look at the `componentWillRender` method in the `cat-button` component for example and implement a similar behavior => log error if label is empty and component is rendered)